construct.js

```diff
@@ -11,11 +12,18 @@ module.exports = (answers, templates, overwrite) => templates.forEach((template)
 
         ensureDirectories(destination);
 
-        if (!fs.existsSync(destination) || overwrite) {
+        if (!fs.existsSync(destination) || overwrite || template.overwrite) {
             const output = replace(template.template, answers, true);
 
             fs.writeFileSync(destination, output, { encoding: 'utf8' });
         }
+        else if(template.update) {
+            const templateObj = JSON.parse(replace(template.template, answers, true));
+            const existing = JSON.parse(fs.readFileSync(destination, 'utf8'));
+
+            const updated = utils.mergeDeep(existing, templateObj);
+            fs.writeFileSync(destination, JSON.stringify(updated, null, 2));
```

Comment:
This and the other `fs.writeFileSync` can be combined into a single call after this `if`/`else if` to keep the write functionality happening in a single place--each `if` should produce a value to `output`, which will be provided to `writeFileSync`.